src/inet/IPAddress.cpp

```diff
@@ -269,6 +269,26 @@ CHIP_ERROR IPAddress::GetIPAddressFromSockAddr(const SockAddr & sockaddr, IPAddr
 
 #endif // CHIP_SYSTEM_CONFIG_USE_SOCKETS || CHIP_SYSTEM_CONFIG_USE_NETWORK_FRAMEWORK
 
+#if CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT
+IPAddress::IPAddress(const otIp6Address & ipv6Addr)
+{
+    memcpy(Addr, ipv6Addr.mFields.m32, sizeof(Addr));
```

Comment:
Maybe we should static_assert somewhere here that sizeof(Addr) == sizeof(ipv6Addr.mFields.m32)?

Reply:
Added some static_assert